feature/google/src/test/java/com/revenuecat/purchases/google/FindBestOfferTest.kt

```diff
@@ -0,0 +1,529 @@
+package com.revenuecat.purchases.google
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.android.billingclient.api.BillingClient
+import com.android.billingclient.api.ProductDetails
+import com.revenuecat.purchases.models.GoogleSubscriptionOption
+import com.revenuecat.purchases.utils.mockPricingPhase
+import com.revenuecat.purchases.utils.mockProductDetails
+import com.revenuecat.purchases.utils.mockSubscriptionOfferDetails
+import org.assertj.core.api.AssertionsForInterfaceTypes.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class FindBestOfferTest {
+    @Test
+    fun `parsed period for P1Y is 365 days`() {
+        val days = parseBillPeriodToDays("P1Y")
+        assertThat(days).isEqualTo(365)
+    }
+
+    @Test
+    fun `parsed period for P2Y is 730 days`() {
+        val days = parseBillPeriodToDays("P2Y")
+        assertThat(days).isEqualTo(730)
+    }
+
```

Comment:
Minor: Maybe it makes sense to extract this into a separate parameterizable test class

```
@RunWith(Parameterized::class)
class PeriodOfferTest(val period: String, val days: Int) {

    companion object {
        @JvmStatic
        @Parameterized.Parameters
        fun data() : Collection<Array<Any>> {
            return listOf(
                arrayOf("P1Y", 365),
                arrayOf("P2Y", 730),
                arrayOf("P3M", 90),
            )
        }
    }

    @Test
    fun `parsed period is correct`() {
        val actual_days = parseBillPeriodToDays(period)
        assertThat(actual_days).isEqualTo(days)
    }
}

```

Also: I don't think we'll encounter any weird double durations, like `P2M1W`, right?

Reply:
TIL... this is really cool!